www/js/diary/services.js

```diff
@@ -432,21 +432,26 @@ angular.module('emission.main.diary.services', ['emission.plugin.logger',
         });
     };
 
-    timeline.readAllConfirmedTrips = function(endTs, deltaTs) {
+    timeline.readAllCompositeTrips = function(endTs, deltaTs) {
       $ionicLoading.show({
         template: $translate.instant('service.reading-server')
       });
       const readPromises = [
-        CommHelper.getRawEntries(["analysis/confirmed_trip"],
+        CommHelper.getRawEntries(["analysis/composite_trip"],
             endTs - deltaTs, endTs, "data.end_ts"),
       ];
       return Promise.all(readPromises)
         .then(([ctList]) => {
             $ionicLoading.hide();
             return ctList.phone_data.map((ct) => {
-                const retVal = ct.data;
-                retVal.id = ct._id["$oid"];
-                return retVal;
+              ct.data._id = ct._id["$oid"];
+              const cp = ct.data.confirmed_place;
+              if (cp) {
+                cp.data._id = cp._id["$oid"];
+                delete ct.data.confirmed_place;
+                ct.data.confirmed_place = cp.data;
```

Comment:
I am not familiar with the `delete` keyword, but how come we are deleting `ct.data.confirmed_place` first, and then immediately using it again? Won't it be deleted?

Reply:
I'll refactor this

Reply:
refactored bd12d1d5ac7d33d63ec2b410360b073a8d1c8497
Way simpler this way, as this is all that was needed